File: NOVEMBER-2025/11-11-2025/Shortest Common Supersequence.cs
class Solution {
    public int minSuperSeq(string s1, string s2) {
        int n = s1.Length;
        int m = s2.Length;
        int[,] dp = new int[n + 1, m + 1];

        for (int i = 1; i <= n; i++) {
            for (int j = 1; j <= m; j++) {
                if (s1[i - 1] == s2[j - 1])
                    dp[i, j] = 1 + dp[i - 1, j - 1];
                else
                    dp[i, j] = Math.Max(dp[i - 1, j], dp[i, j - 1]);
            }
        }

        int lcsLength = dp[n, m];
        return n + m - lcsLength;
    }
}

File: NOVEMBER-2025/11-11-2025/Shortest Common Supersequence.js
/**
 * @param {string} s1
 * @param {string} s2
 * @returns {number}
 */
class Solution {
    minSuperSeq(s1, s2) {
        const n = s1.length;
        const m = s2.length;
        const dp = Array.from({ length: n + 1 }, () => Array(m + 1).fill(0));

        // Compute LCS
        for (let i = 1; i <= n; i++) {
            for (let j = 1; j <= m; j++) {
                if (s1[i - 1] === s2[j - 1]) {
                    dp[i][j] = 1 + dp[i - 1][j - 1];
                } else {
                    dp[i][j] = Math.max(dp[i - 1][j], dp[i][j - 1]);
                }
            }
        }

        const lcsLength = dp[n][m];
        return n + m - lcsLength;
    }
}